lang/csharp/src/apache/main/Util/Date.cs

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+
+namespace Avro.Util
+{
+    /// <summary>
+    /// The 'date' logical type.
+    /// </summary>
+    public class Date : LogicalUnixEpochType<DateTime>
+    {
+        /// <summary>
+        /// The logical type name for Date.
+        /// </summary>
+        public static readonly string LogicalTypeName = "date";
+
+        /// <summary>
+        /// Initializes a new Date logical type.
+        /// </summary>
+        public Date() : base(LogicalTypeName)
```

Comment:
I think you want `schema` to be passed as a constructor argument, not as a parameter in the methods below. Either that or have everything purely functional. Note: again, haven't analysed everything super closely, just pointing out something that seems wrong.

Reply:
Yes, I felt this approach fitted with the surrounding code, plus I took a look at the Java approach at the time. The `LogicalType` is used like a flyweight, since we have instances of the schemas and their containing properties I've reduced the need to have an instance of the LogicalType too.

Reply:
ok - i better understand what you're doing now, i think it's fine.